### DISCUSSION TITLE
Set non exported values

### DISCUSSION THREAD

**Discussion by eldegende** [Ark2000/PankuConsole#167] [Q&A] [answered]
How to set/modify non-exported values, such as position or scale of an object in console?
I've tried setting values like `current.player.position.x = 0`, but it returns `Expected '='` error

**Comment by Ark2000** [comment, marked as answer]
unfortunately you can not directly modify a non-object's property by a single expression, but you can achieve this by a wrapper function like this:

```gdscript
func set_anything(variable, property:String, val):
	if typeof(variable) == TYPE_VECTOR2:
		if property == "x":
			variable.x = val
		elif property == "y":
			variable.y = val
	return variable
```

now, you can change a vector2's property by

```gdscript
# a is a sprite2d
current.a.set("position", current.set_anything(current.a.position, "x", 100))
```

this is indeed inconvenient, need more work on command parser in the future work.

**Reply by eldegende** [reply]
Ok, thanks!

**Comment by pewcworrell** [comment]
I made a change to improve this. I replaced the checks for 
```gdscript
d.usage == PROPERTY_USAGE_SCRIPT_VARIABLE | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_STORAGE
```
with
```gdscript
(d.usage == PROPERTY_USAGE_SCRIPT_VARIABLE | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_STORAGE) or (d.usage == PROPERTY_USAGE_SCRIPT_VARIABLE | PROPERTY_USAGE_NO_INSTANCE_STATE)
```
in exporter_2.gd and gdexprenv.gd. Then each property that I want to edit in game, but not in editor, I add the following export to:
```gdscript
@export_custom(PROPERTY_HINT_NONE, "", PROPERTY_USAGE_SCRIPT_VARIABLE | PROPERTY_USAGE_NO_INSTANCE_STATE)
```
Works like a charm.

**Reply by pewcworrell** [reply]
I may end up adjusting the flags if I find it's saving to a scene file, but I don't think it will since I have no way to change it from default and the default isn't changed. I also contemplated making use of a deprecated flag like PROPERTY_USAGE_DEFERRED_SET_RESOURCE, or making a custom hint_string, but this works fine. I searched the engine source and [this](https://github.com/godotengine/godot/blob/f0f5319b0b5b56db9f8023ac4132088df9e86b9e/scene/resources/packed_scene.cpp#L798) is the only place I could find any usage of `PROPERTY_USAGE_DEFERRED_SET_RESOURCE`
